### ISSUE TITLE
Error using OpenSSL 1.0.2s

### ISSUE THREAD

**Issue by AndyPeakSW** [conan-community/community#322]
### Description of Problem, Request, or Question
Error is generated when attempting to download OpenSSL using Conan:
```
ERROR: OpenSSL/1.0.2s@conan/stable: Error in source() method, line 170
	tools.get(url, sha256=sha256[self.version])
	NotFoundException: Not found: https://www.openssl.org/source/old/1.0.2openssl-1.0.2s.tar.gz
```
There used to be no errors with this package, but it started generating an error on 2 Apr 2020, whereas it was available on 1 Apr 2020 at 10:57 GMT

It appears that `https://www.openssl.org/source/old/1.0.2openssl-1.0.2s.tar.gz` does not exist anymore. According to https://www.openssl.org/source/old/1.0.2/, the URL should be `https://www.openssl.org/source/old/1.0.2/openssl-1.0.2s.tar.gz`

Changing expression on line 168/9 in OpenSSL/1.0.2s/conan/stable/export/conanfile.py from:
```
            url = url.replace("https://www.openssl.org/source/",
                              "https://www.openssl.org/source/old/%s" % self._full_version.base)
```
to add "/" after "%s"
```
            url = url.replace("https://www.openssl.org/source/",
                              "https://www.openssl.org/source/old/%s/" % self._full_version.base)
```
Modifies the URL from `https://www.openssl.org/source/old/1.0.2openssl-1.0.2s.tar.gz` to `https://www.openssl.org/source/old/1.0.2/openssl-1.0.2s.tar.gz` and appears to resolve the problem.

It looks like the OpenSSL tarball 1.0.2openssl-1.0.2s.tar.gz has been removed in preference for the tarballs in the 1.0.2 sub-folder
### Package Details (Include if Applicable)
* Package Name/Version: **openssl/1.0.2s**
* Operating System: **Linux**
* Operation System Version: **Ubuntu 18.04**

### Steps to reproduce (Include if Applicable)
Include a package that uses OpenSSL

**Comment by SSE4** [comment]
hi @AndyPeakSW 
pls try switching to the `openssl/1.0.2s` package (without username/channel)

**Comment by AndyPeakSW** [comment]
Hi,
The package without username/channel could not be found.
```
conanfile: configuring project settings ...
OpenSSL/1.0.2s: Not found in local cache, looking in remotes...
OpenSSL/1.0.2s: Trying with 'conan-center'...
ERROR: Failed requirement 'OpenSSL/1.0.2s' from 'conanfile.py (None/0.1)'
ERROR: Unable to find 'OpenSSL/1.0.2s' in remotes
```

Specifying the "OpenSSL/1.0.2s@conan/stable", downloads the package okay but fails in the configuration of the package
```
conanfile: configuring project settings ...
OpenSSL/1.0.2s@conan/stable: Not found in local cache, looking in remotes...
OpenSSL/1.0.2s@conan/stable: Trying with 'conan-center'...
Downloading conanmanifest.txt

Downloading conanfile.py

OpenSSL/1.0.2s@conan/stable: Downloaded recipe revision 0

....

OpenSSL/1.0.2s@conan/stable: Configuring sources in /home/***/.conan/data/OpenSSL/1.0.2s/conan/stable/source
ERROR: OpenSSL/1.0.2s@conan/stable: Error in source() method, line 170
	tools.get(url, sha256=sha256[self.version])
	NotFoundException: Not found: https://www.openssl.org/source/old/1.0.2openssl-1.0.2s.tar.gz

```

**Comment by Croydon** [comment]
Please note that it is lower-case `openssl/1.0.2s` not upper-case `OpenSSL/1.0.2s`

`OpenSSL/1.0.2s@conan/stable` is deprecated. Please migrate away from it as soon as possible

**Comment by AndyPeakSW** [comment]
The lower-case `openssl/1.0.2s` ends up with a zlib error:
```
ERROR: zlib/1.2.11: Error in package_info() method, line 166
	self.cpp_info.names["cmake_find_package"] = "ZLIB"
	TypeError: '_CppInfo' object does not support item assignment
```
The reason why `OpenSSL/1.0.2s@conan/stable` is being used is because it is being pulled in by `paho-cpp/1.0.1@conan/stable`, which is what we need to use.

Therefore it would be useful to have a fix for `OpenSSL/1.0.2s@conan/stable`

**Comment by danimtb** [comment]
Sorry @AndyPeakSW but we are not maintaining the recipes under this organization anymore as the new process to contribute recipes to ConanCenter is using https://github.com/conan-io/conan-center-index

IT is recommended to switch to the references without user/channel like `openssl/1.0.2s`. You can find the paho cop library under the reference `paho-mqtt-cpp/1.0.1`.

The zlib error you are facing is caused due to the recent updates of the recipes with new features not supported by old versions of Conan. Please update your Conan client to the latest version
